src/R/Editor/Impl/Completion/Providers/FilesCompletionProvider.cs

```diff
@@ -22,14 +22,22 @@ namespace Microsoft.R.Editor.Completion.Providers {
     /// Provides list of files and folder in the current directory
     /// </summary>
     internal sealed class FilesCompletionProvider : IRCompletionListProvider {
+        enum Mode {
+            WorkingDirectory,
+            UserDirectory,
+            Other
+        }
+
         private readonly IImagesProvider _imagesProvider;
         private readonly IRInteractiveWorkflow _workflow;
         private readonly IGlyphService _glyphService;
 
-        private Task<string> _userDirectoryFetchingTask;
-        private string _directory;
-        private string _cachedUserDirectory;
-        private bool _forceR; // for tests
+        private readonly string _enteredDirectory;
+        private readonly bool _forceR; // for tests
+
+        private readonly Task<string> _task;
+        private readonly Mode _mode = Mode.Other;
```

Comment:
It is set in ctor and then used down in `GetEntries`. I'd like to split cases going forward and derive to separate classes from base rather than use conditional by mode. But that would be much larger change.